### DISCUSSION TITLE
Remove `FileNotFound` error

### DISCUSSION THREAD

**Discussion by rylev** [zip-rs/zip-old#341] [Feature Design]
Currently the `FileNotFound` error does not really represent an error case so much as the user has asked for a file that doesn't exist. This is normally represented as a `None` in an option. I think we should return `ZipResult<Option<ZipFile>>` from `ZipArchive::by_index` and `ZipArchive::by_name`

**Comment by Plecra** [comment]
I think the error type could do with some TLC. Is there any reason we shouldn't replace the `&'static str`s with some more descriptive variants?

**Comment by rylev** [comment]
I'd like to open a new issue to address that. This should remain just for the FileNotFound case.
